BEIMA.Backend.Test/AuthFunctions/LoginTest.cs

```diff
@@ -0,0 +1,165 @@
+﻿using BEIMA.Backend.AuthFunctions;
+using BEIMA.Backend.Models;
+using BEIMA.Backend.MongoService;
+using JWT.Builder;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using static BEIMA.Backend.Test.RequestFactory;
+using BCryptNet = BCrypt.Net.BCrypt;
+
+namespace BEIMA.Backend.Test.AuthFunctions
+{
+    [TestFixture]
+    public class LoginTest : UnitTestBase
+    {
+        [Test]
+        public async Task NullPostBody_Login_ReturnsNull()
+        {
+            // ARRANGE
+            var body = TestData._testNullLoginRequest;
+            var request = CreateHttpRequest(RequestMethod.POST, body: body);
+            var logger = (new LoggerFactory()).CreateLogger("Testing");
+
+            // ACT
+            var token = ((ObjectResult)await Login.Run(request, logger)).Value?.ToString();
+
+            // ASSERT
+            Assert.IsNotNull(token);
+            Assert.That(token, Is.EqualTo("Invalid credentials."));
+        }
+
+        [Test]
+        public async Task InvalidParameters_Login_ReturnsNull()
+        {
+            // ARRANGE
+            var body = TestData._testNullKeysLoginRequest;
+            var request = CreateHttpRequest(RequestMethod.POST, body: body);
+            var logger = (new LoggerFactory()).CreateLogger("Testing");
+
+            // ACT
+            var token = ((ObjectResult) await Login.Run(request, logger)).Value?.ToString();
+
+            // ASSERT
+            Assert.IsNotNull(token);
+            Assert.That(token, Is.EqualTo("Invalid credentials."));
+        }
+
+        [Test]
+        public async Task ValidParametersNoFilteredUser_Login_ReturnsNull()
+        {
+            // ARRANGE
+            Mock<IMongoConnector> mockDb = new Mock<IMongoConnector>();
+            mockDb.Setup(mock => mock.GetFilteredUsers(It.IsAny<FilterDefinition<BsonDocument>>()))
+                  .Returns(new List<BsonDocument>())
+                  .Verifiable();
+
+            var body = TestData._testValidKeysLoginRequest;
+            var request = CreateHttpRequest(RequestMethod.POST, body: body);
+            var logger = (new LoggerFactory()).CreateLogger("Testing");
+
+            // ACT
+            var token = ((ObjectResult)await Login.Run(request, logger)).Value?.ToString();
+
+            // ASSERT
```

Comment:
Don't forget to add the asserts to verify whether or not the mock database method ran.